### ISSUE TITLE
Build warnings

### ISSUE THREAD

**Issue by SpacyRicochet** [openfoodfacts/openfoodfacts-ios#930]
Running the project for the first time in Xcode 12.5 causes a couple of compiler warnings to appear.

* Use of `AnyObject` conformance instead of the deprecated `class`.
* Explicitly put completion handlers in the parameter.

These seem easy and worthwhile to fix without issue.

**Comment by teolemon** [comment]
thanks @SpacyRicochet :-)
Would you have a go at it ?

**Comment by SpacyRicochet** [comment]
Sure. Though I just figured out I made a mistake while creating this issue. I was building in Xcode 12.5. The same warnings and fixes for them still apply, so I just updated the title to avoid mentioning any specifics. 

Compiling for Xcode 13 proved to be more cumbersome. Some of the Carthage dependencies don't build well. I started with Nimble, which just required a small update… however, Realm is currently at v3.9.1 and the latest version is either 10.16.0 or 5.5.2 — I'm not entirely sure what the deal is with that — so I haven't attempted an upgrade of that yet.

Continuing with Xcode 12.5 for now, but Realm might need a proper upgrade to continue proper development.
